*** a/frontend_qt/mainwindow.cpp	2023-12-18 10:44:47.000000000 +0100
--- b/frontend_qt/mainwindow.cpp	2024-08-04 15:53:49.125638832 +0200
***************
*** 28,33 ****
--- 28,34 ----
  #include <QClipboard>
  #include <QMimeData>
  #include <QColor>
+ #include <QtGlobal>
  #include <QListView>
  #include <QShortcut>
  #include <QMenu>
***************
*** 3507,3516 ****
      m_copyTIFShortcut->setEnabled(enabled);
  }
  
! void MainWindow::copy_to_clipboard(const QString &filename, const QString& name, const char *mimeType)
  {
      QClipboard *clipboard = QGuiApplication::clipboard();
  
      if (!m_bc.bc.save_to_file(filename)) {
          return;
      }
--- 3508,3518 ----
      m_copyTIFShortcut->setEnabled(enabled);
  }
  
! void MainWindow::copy_to_clipboard(const QString &in_filename, const QString& name, const char *mimeType)
  {
      QClipboard *clipboard = QGuiApplication::clipboard();
  
+     auto filename = QString::fromLocal8Bit(qgetenv("XDG_RUNTIME_DIR")).append("/app/uk.org.zint.zint-qt/").append(in_filename);
      if (!m_bc.bc.save_to_file(filename)) {
          return;
      }
